gnucash/gnome-utils/dialog-preferences.c

```diff
@@ -1236,6 +1236,9 @@ gnc_preferences_dialog_create(void)
     // Set the style context for this dialog so it can be easily manipulated with css
     gnc_widget_set_style_context (GTK_WIDGET(dialog), "GncPreferenceDialog");
 
+    /* parent */
+    gtk_window_set_transient_for (GTK_WINDOW(dialog), GTK_WINDOW(gnc_ui_get_main_window (NULL)));
```

Comment:
See if you can get the calling functions to pass a parent widget like for the others below.